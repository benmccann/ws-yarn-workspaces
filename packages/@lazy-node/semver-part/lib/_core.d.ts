export declare function _versionUnsafe(part: string, defaultValue?: string): string;
export declare function _version(part1: string): string;
export declare function _part(part1: string, part2: string): [string, string];
export declare function partsToVersion(parts: string[]): string;
export declare function versionToParts(version: string): string[];
