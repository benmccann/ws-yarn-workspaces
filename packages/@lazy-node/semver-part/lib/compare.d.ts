export type { Operator, Options } from 'semver';
import { Operator, Options } from 'semver';
export declare type IOptionsOrLoose = boolean | Options;
export declare type ICompareReturnType = 1 | 0 | -1;
export declare function compare(part1: string, part2: string, optionsOrLoose?: boolean | Options): 0 | 1 | -1;
export declare function eq(part1: string, part2: string, optionsOrLoose?: boolean | Options): boolean;
export declare function neq(part1: string, part2: string, optionsOrLoose?: boolean | Options): boolean;
export declare function gt(part1: string, part2: string, optionsOrLoose?: boolean | Options): boolean;
export declare function gte(part1: string, part2: string, optionsOrLoose?: boolean | Options): boolean;
export declare function lt(part1: string, part2: string, optionsOrLoose?: boolean | Options): boolean;
export declare function lte(part1: string, part2: string, optionsOrLoose?: boolean | Options): boolean;
export declare function cmp(part1: string, operator: Operator, part2: string, optionsOrLoose?: boolean | Options): boolean;
