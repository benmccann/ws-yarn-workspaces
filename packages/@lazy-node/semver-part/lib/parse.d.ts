import { IOptionsOrLoose, ICompareReturnType } from './compare';
export declare type IParseVersionsFindIndex = 0 | 1 | 2;
export declare function parseVersions(versionOld: string, versionNew: string): {
    versionOld: string;
    versionNew: string;
    partsOld: string[];
    partsNew: string[];
    index: IParseVersionsFindIndex;
};
export declare function parseVersionsAndCompare(versionOld: string, versionNew: string, optionsOrLoose?: IOptionsOrLoose): {
    comp: ICompareReturnType;
    versionOld: string;
    versionNew: string;
    partsOld: string[];
    partsNew: string[];
    index: IParseVersionsFindIndex;
};
