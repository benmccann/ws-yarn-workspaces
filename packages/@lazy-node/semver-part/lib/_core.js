"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.versionToParts = exports.partsToVersion = exports._part = exports._version = exports._versionUnsafe = void 0;
function _versionUnsafe(part, defaultValue) {
    var _a;
    return `0.0.${(_a = part !== null && part !== void 0 ? part : defaultValue) !== null && _a !== void 0 ? _a : 0}`;
}
exports._versionUnsafe = _versionUnsafe;
function _version(part1) {
    if (/^\d+\.\d+\./.test(part1)) {
        return part1;
    }
    else if (/^\d+\.(?:[^\.]+)$/.test(part1)) {
        return '0.' + part1;
    }
    return _versionUnsafe(part1);
}
exports._version = _version;
function _part(part1, part2) {
    if (/^\d+\.\d+\./.test(part1) && /^\d+\.\d+\./.test(part2)) {
        return [part1, part2];
    }
    else if (/^\d+\.(?:[^\.]+)$/.test(part1) && /^\d+\.(?:[^\.]+)$/.test(part2)) {
        return ['0.' + part1, '0.' + part2];
    }
    return [_versionUnsafe(part1), _versionUnsafe(part2)];
}
exports._part = _part;
function partsToVersion(parts) {
    return parts.join('.');
}
exports.partsToVersion = partsToVersion;
function versionToParts(version) {
    let parts = version.split('.');
    if (parts.length > 3) {
        return [parts[0], parts[1], partsToVersion(parts.slice(2))];
    }
    return parts;
}
exports.versionToParts = versionToParts;
//# sourceMappingURL=_core.js.map