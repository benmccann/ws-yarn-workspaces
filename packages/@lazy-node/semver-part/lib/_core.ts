
export function _versionUnsafe(part: string, defaultValue?: string)
{
	return `0.0.${part ?? defaultValue ?? 0}`;
}

export function _version(part1: string)
{
	if (/^\d+\.\d+\./.test(part1))
	{
		return part1
	}
	else if (/^\d+\.(?:[^\.]+)$/.test(part1))
	{
		return '0.' + part1
	}

	return _versionUnsafe(part1)
}

export function _part(part1: string, part2: string): [string, string]
{
	if (/^\d+\.\d+\./.test(part1) && /^\d+\.\d+\./.test(part2))
	{
		return [part1, part2]
	}
	else if (/^\d+\.(?:[^\.]+)$/.test(part1) && /^\d+\.(?:[^\.]+)$/.test(part2))
	{
		return ['0.' + part1, '0.' + part2]
	}

	return [_versionUnsafe(part1), _versionUnsafe(part2)]
}

export function partsToVersion(parts: string[])
{
	return parts.join('.')
}

export function versionToParts(version: string): string[]
{
	let parts = version.split('.')

	if (parts.length > 3)
	{
		return [parts[0], parts[1], partsToVersion(parts.slice(2))]
	}

	return parts
}
