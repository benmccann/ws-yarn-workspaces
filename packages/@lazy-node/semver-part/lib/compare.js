"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.tryCompare = exports.cmp = exports.lte = exports.lt = exports.gte = exports.gt = exports.neq = exports.eq = exports.compare = void 0;
/**
 * Created by user on 2020/6/11.
 */
const _core_1 = require("./_core");
const compare_1 = __importDefault(require("semver/functions/compare"));
const cmp_1 = __importDefault(require("semver/functions/cmp"));
function compare(part1, part2, optionsOrLoose) {
    return compare_1.default(..._core_1._part(part1, part2), optionsOrLoose);
}
exports.compare = compare;
function eq(part1, part2, optionsOrLoose) {
    return compare(part1, part2, optionsOrLoose) === 0;
}
exports.eq = eq;
function neq(part1, part2, optionsOrLoose) {
    return compare(part1, part2, optionsOrLoose) !== 0;
}
exports.neq = neq;
function gt(part1, part2, optionsOrLoose) {
    return compare(part1, part2, optionsOrLoose) > 0;
}
exports.gt = gt;
function gte(part1, part2, optionsOrLoose) {
    return compare(part1, part2, optionsOrLoose) >= 0;
}
exports.gte = gte;
function lt(part1, part2, optionsOrLoose) {
    return compare(part1, part2, optionsOrLoose) < 0;
}
exports.lt = lt;
function lte(part1, part2, optionsOrLoose) {
    return compare(part1, part2, optionsOrLoose) <= 0;
}
exports.lte = lte;
function cmp(part1, operator, part2, optionsOrLoose) {
    const [v1, v2] = _core_1._part(part1, part2);
    return cmp_1.default(v1, operator, v2, optionsOrLoose);
}
exports.cmp = cmp;
function tryCompare(v1, v2, optionsOrLoose) {
    try {
        return compare(v1, v2, optionsOrLoose);
    }
    catch (e) { }
}
exports.tryCompare = tryCompare;
//# sourceMappingURL=compare.js.map