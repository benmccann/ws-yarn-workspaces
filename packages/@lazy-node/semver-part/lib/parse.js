"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseVersionsAndCompare = exports.parseVersions = void 0;
const _core_1 = require("./_core");
const compare_1 = require("./compare");
function parseVersions(versionOld, versionNew) {
    const partsNew = _core_1.versionToParts(versionNew);
    const partsOld = _core_1.versionToParts(versionOld);
    let index = partsNew.findIndex((part, i) => part !== partsOld[i]);
    index = index >= 0 ? index : partsNew.length;
    return {
        versionOld,
        versionNew,
        partsOld,
        partsNew,
        index,
    };
}
exports.parseVersions = parseVersions;
function parseVersionsAndCompare(versionOld, versionNew) {
    const data = parseVersions(versionOld, versionNew);
    let comp = compare_1.tryCompare(data.partsNew[data.index], data.partsOld[data.index]);
    return {
        ...data,
        comp,
    };
}
exports.parseVersionsAndCompare = parseVersionsAndCompare;
//# sourceMappingURL=parse.js.map