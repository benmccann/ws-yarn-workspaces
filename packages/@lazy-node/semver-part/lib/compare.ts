/**
 * Created by user on 2020/6/11.
 */
import { _part } from './_core';

import _compare from 'semver/functions/compare';

import _eq from 'semver/functions/eq';
import _neq from 'semver/functions/neq';

import _gt from 'semver/functions/gt';
import _gte from 'semver/functions/gte';

import _lt from 'semver/functions/lt';
import _lte from 'semver/functions/lte';

import _cmp from 'semver/functions/cmp';

export type { Operator, Options } from 'semver';
import { Operator, Options } from 'semver';

export type IOptionsOrLoose = boolean | Options
export type ICompareReturnType = 1 | 0 | -1

export function compare(part1: string, part2: string, optionsOrLoose?: boolean | Options)
{
	return _compare(..._part(part1, part2), optionsOrLoose as any)
}

export function eq(part1: string, part2: string, optionsOrLoose?: boolean | Options)
{
	return compare(part1, part2, optionsOrLoose as any) === 0
}

export function neq(part1: string, part2: string, optionsOrLoose?: boolean | Options)
{
	return compare(part1, part2, optionsOrLoose as any) !== 0
}

export function gt(part1: string, part2: string, optionsOrLoose?: boolean | Options)
{
	return compare(part1, part2, optionsOrLoose as any) > 0
}

export function gte(part1: string, part2: string, optionsOrLoose?: boolean | Options)
{
	return compare(part1, part2, optionsOrLoose as any) >= 0
}

export function lt(part1: string, part2: string, optionsOrLoose?: boolean | Options)
{
	return compare(part1, part2, optionsOrLoose as any) < 0
}

export function lte(part1: string, part2: string, optionsOrLoose?: boolean | Options)
{
	return compare(part1, part2, optionsOrLoose as any) <= 0
}

export function cmp(part1: string, operator: Operator, part2: string, optionsOrLoose?: boolean | Options)
{
	const [v1, v2] = _part(part1, part2)

	return _cmp(v1, operator, v2, optionsOrLoose)
}
