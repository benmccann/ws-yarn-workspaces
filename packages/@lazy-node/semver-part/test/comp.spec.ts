import { inspect } from "util";
import { versionToParts, partsToVersion } from '..';
import { Options } from 'semver';
import { ICompareReturnType, compare } from '../lib/compare';

describe(`compare`, () =>
{

	([
		[
			['3-alpha', '3-alpha'],
		],
		[
			['3-alpha', '3-prealpha'],
		],
		[
			['4-alpha', '3-prealpha'],
		],
		[
			['3-alpha', '4-prealpha'],
		],
		[
			['7.9.3-alpha', '7.9.3-alpha'],
		],
		[
			['7.9.3-alpha', '7.9.3-prealpha'],
		],
		[
			['7.9.4-alpha', '7.9.3-prealpha'],
		],
		[
			['7.9.3-alpha', '7.9.4-prealpha'],
		],
		[
			['9.3-alpha', '9.3-alpha'],
		],
		[
			['9.3-alpha', '9.3-prealpha'],
		],
		[
			['9.4-alpha', '9.3-prealpha'],
		],
		[
			['9.3-alpha', '9.4-prealpha'],
		],
		[
			['3', '3'],
		],
		[
			['4', '3'],
		],
		[
			['3', '4'],
		],
	] as [[string, string, (boolean | Options)?], ICompareReturnType?][]).forEach(tastCase => {

		test(inspect(tastCase[0]), () =>
		{

			let actual = compare(...tastCase[0]);
			let expected = tastCase[1];

			if (typeof expected !== 'undefined')
			{
				expect(actual).toStrictEqual(expected);
			}

			expect(actual).toMatchSnapshot();
		});

	})

})
