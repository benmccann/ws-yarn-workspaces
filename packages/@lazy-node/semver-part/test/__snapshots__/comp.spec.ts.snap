// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compare [ '3', '3' ] 1`] = `0`;

exports[`compare [ '3', '4' ] 1`] = `-1`;

exports[`compare [ '3-alpha', '3-alpha' ] 1`] = `0`;

exports[`compare [ '3-alpha', '3-prealpha' ] 1`] = `-1`;

exports[`compare [ '3-alpha', '4-prealpha' ] 1`] = `-1`;

exports[`compare [ '4', '3' ] 1`] = `1`;

exports[`compare [ '4-alpha', '3-prealpha' ] 1`] = `1`;

exports[`compare [ '7.9.3-alpha', '7.9.3-alpha' ] 1`] = `0`;

exports[`compare [ '7.9.3-alpha', '7.9.3-prealpha' ] 1`] = `-1`;

exports[`compare [ '7.9.3-alpha', '7.9.4-prealpha' ] 1`] = `-1`;

exports[`compare [ '7.9.4-alpha', '7.9.3-prealpha' ] 1`] = `1`;

exports[`compare [ '9.3-alpha', '9.3-alpha' ] 1`] = `0`;

exports[`compare [ '9.3-alpha', '9.3-prealpha' ] 1`] = `-1`;

exports[`compare [ '9.3-alpha', '9.4-prealpha' ] 1`] = `-1`;

exports[`compare [ '9.4-alpha', '9.3-prealpha' ] 1`] = `1`;
